### ISSUE TITLE
[FEA] Default executorEnv for UCX

### ISSUE THREAD

**Issue by abellina** [NVIDIA/spark-rapids#1929]
There are at least two boilerplate settings that need to be set for UCX: disabling signal catching, and disabling the memtype cache (though in later versions of UCX the second one here is default off). There are more settings, especially for the wakeup protocol (i.e. we memory mapped IO within a machine has been shown to cause hangs, when used in conjunction with wakeup).

When we launch JUCX it is too late for it to change these settings and it would require bigger changes in UCX to make it happen. Instead, I was thinking of potentially mutating `executorEnv` (which looks to be `private[spark]` right now in SparkContext). 

This issue is to figure out if we can mutate `executorEnv` in every environment (i.e. it works in Standalone, but @tgravescs pointed out that Yarn client-mode may not).

If this does work, at least for the majority of environments users will be working under, I think it may be sensible to add the code. Additionally, a Spark issue could be filed for the cases where it doesn't.

**Comment by sameerz** [comment]
Preference is to put up a SPIP for Spark plugins to modify the executorEnv if so desired.  

@tgravescs is this something that is worth pursuing without going the SPIP route first?

**Comment by tgravescs** [comment]
yes I think we should prototype and see how much changes. If its minor change to the plugin api which is developer api shouldn't need a spip
